clar2wasm/src/linker.rs

```diff
@@ -4036,6 +4038,114 @@ fn link_principal_of_fn(linker: &mut Linker<ClarityWasmContext>) -> Result<(), E
+             name_offset: i32,
+             name_length: i32,
+             value_offset: i32,
+             _value_length: i32| {
+                let memory = caller
+                    .get_export("memory")
+                    .and_then(|export| export.into_memory())
+                    .ok_or(Error::Wasm(WasmError::MemoryNotFound))?;
+
+                let epoch = caller.data_mut().global_context.epoch_id;
+
+                // Get constant name from the memory.
+                let const_name =
+                    read_identifier_from_wasm(memory, &mut caller, name_offset, name_length)?;
+                let cname = ClarityName::from(const_name.as_str());
+
+                // Get constant value type.
+                let value_ty = caller
+                    .data()
+                    .contract_analysis
+                    .ok_or(Error::Wasm(WasmError::DefinesNotFound))?
+                    .get_variable_type(const_name.as_str())
+                    .ok_or(Error::Wasm(WasmError::DefinesNotFound))?;
+
+                let value =
+                    read_from_wasm_indirect(memory, &mut caller, value_ty, value_offset, epoch)?;
+
+                // Insert constant name and expression value into a persistent data structure.
+                caller
+                    .data_mut()
+                    .contract_context_mut()?
+                    .variables
+                    .insert(cname, value);
+
+                Ok(())
+            },
+        )
+        .map(|_| ())
+        .map_err(|e| {
+            Error::Wasm(WasmError::UnableToLinkHostFunction(
+                "save_constant".to_string(),
+                e,
+            ))
+        })
+}
+
+fn link_load_constant_fn(linker: &mut Linker<ClarityWasmContext>) -> Result<(), Error> {
+    linker
+        .func_wrap(
+            "clarity",
+            "load_constant",
+            |mut caller: Caller<'_, ClarityWasmContext>,
+             name_offset: i32,
+             name_length: i32,
+             value_offset: i32,
+             _value_length: i32| {
+                let memory = caller
+                    .get_export("memory")
+                    .and_then(|export| export.into_memory())
+                    .ok_or(Error::Wasm(WasmError::MemoryNotFound))?;
+
+                // Read constant name from the memory.
+                let const_name =
+                    read_identifier_from_wasm(memory, &mut caller, name_offset, name_length)?;
+
+                // Constant value
+                let value = caller
+                    .data()
+                    .contract_context()
+                    .variables
+                    .get(&ClarityName::from(const_name.as_str()))
+                    // TODO: create a new WasmError to handle the constant not found issue
+                    .ok_or(Error::Wasm(WasmError::WasmGeneratorError(
+                        "Constant not found on MARF".to_string(),
+                    )))?
```

Comment:
I think you could use:
```rust
.ok_or(CheckErrors::UndefinedVariable(const_name.to_string()))?
```
That's the error that the type-checker would throw.

Reply:
Changed. Tks!